tests/test_dagviz.py

```diff
@@ -24,9 +37,54 @@ def flow_image(flow):
     return flow.render_dag()
 
 
-def get_pydot_attributes(index, dot):
-    """Helper function to get attributes from pydot graph given index"""
-    return dot.get_subgraphs()[index].get_nodes()[0].get_attributes()
+@pytest.fixture
+def flow_graph(flow):
+    return flow._deriver.export_dag()
+
+
+@pytest.fixture
+def flow_dot(flow_graph):
+    return dagviz.dot_from_graph(flow_graph)
+
+
+def nodes_by_name_from_dot(dot):
+    return {
+        node.get_name(): node
+        for subgraph in dot.get_subgraphs()
+        for node in subgraph.get_nodes()
+    }
+
+
+def test_dag_size(flow_graph):
+    assert len(flow_graph.nodes) == 8
+
+
+def test_dot_properties(flow_dot):
+    nodes = nodes_by_name_from_dot(flow_dot)
+    assert set(nodes.keys()) == {
+        # pydot puts quotes around the name if it contains square brackets. (However,
+        # these quotes are not visible when the graph is rendered as an image.)
+        '"first_name[0]"',
+        '"first_name[1]"',
+        "last_name",
+        '"first_name[0]"',
```

Comment:
nit: you can get rid of this repeated first name

Reply:
Thanks!